test/cli/install/bun-upgrade.test.ts

```diff
@@ -0,0 +1,112 @@
+import { spawn, spawnSync } from "bun";
```

Comment:
for all `bun upgrade` tests that actually invoke the upgrader, you need to copy the bun binary somewhere else then run it, or else bun will override itself, and you'll be testing the latest release.

Reply:
Hey @paperdave, I believe this commit (https://github.com/oven-sh/bun/pull/7784/commits/e0e76c436aaf5825947ae85a5661b1f097342d51) resolves this. Please let me know if not :)